substrate/frame/staking-async/src/pallet/impls.rs

```diff
@@ -1958,4 +1959,23 @@ impl<T: Config> Pallet<T> {
 
 		Ok(())
 	}
+
+	/// Invariants:
+	/// - No cancelled slashes should exist for eras before the active era.
+	/// - No unapplied slashes should exist for eras before the active era.
+	fn check_deferred_slashes() -> Result<(), TryRuntimeError> {
+		let active_era = session_rotation::Rotator::<T>::active_era();
+
+		// Ensure no unapplied slashes exist before the active era
+		for (era, _, _) in UnappliedSlashes::<T>::iter() {
+			ensure!(era >= active_era, "Found unapplied slashes for era before active era");
```

Comment:
```suggestion
			ensure!(era < active_era, "Found unapplied slashes for era before active era");
```

Reply:
Its correct I think.

So the era in unapplied slashes must always be equal or greater than active era. If its less, it means the slash era passed but the slash for that era was not applied.